src/blockdata/opcodes.rs

```diff
@@ -652,48 +652,99 @@ impl fmt::Debug for All {
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ClassifyContext {
+    /// Opcode used in tapscript context
+    TapScript,
+    /// Opcode used in legacy context
+    Legacy,
+}
+
 impl All {
     /// Classifies an Opcode into a broad class
     #[inline]
-    pub fn classify(self) -> Class {
-        // 17 opcodes
-        if self == all::OP_VERIF || self == all::OP_VERNOTIF ||
-           self == all::OP_CAT || self == all::OP_SUBSTR ||
-           self == all::OP_LEFT || self == all::OP_RIGHT ||
-           self == all::OP_INVERT || self == all::OP_AND ||
-           self == all::OP_OR || self == all::OP_XOR ||
-           self == all::OP_2MUL || self == all::OP_2DIV ||
-           self == all::OP_MUL || self == all::OP_DIV || self == all::OP_MOD ||
-           self == all::OP_LSHIFT || self == all::OP_RSHIFT {
-            Class::IllegalOp
-        // 11 opcodes
-        } else if self == all::OP_NOP ||
-                  (all::OP_NOP1.code <= self.code &&
-                   self.code <= all::OP_NOP10.code) {
-            Class::NoOp
-        // 75 opcodes
-        } else if self == all::OP_RESERVED || self == all::OP_VER || self == all::OP_RETURN ||
-                  self == all::OP_RESERVED1 || self == all::OP_RESERVED2 ||
-                  self.code >= all::OP_RETURN_186.code {
-            Class::ReturnOp
-        // 1 opcode
-        } else if self == all::OP_PUSHNUM_NEG1 {
-            Class::PushNum(-1)
-        // 16 opcodes
-        } else if all::OP_PUSHNUM_1.code <= self.code &&
-                  self.code <= all::OP_PUSHNUM_16.code {
-            Class::PushNum(1 + self.code as i32 - all::OP_PUSHNUM_1.code as i32)
-        // 76 opcodes
-        } else if self.code <= all::OP_PUSHBYTES_75.code {
-            Class::PushBytes(self.code as u32)
-        // 60 opcodes
-        } else {
-            Class::Ordinary(Ordinary::try_from_all(self).unwrap())
+    pub fn classify(self, ctx: ClassifyContext) -> Class {
+        // 3 opcodes
+        match ctx {
+            ClassifyContext::TapScript => // 3 opcodes
+                if self == all::OP_VERIF || self == all::OP_VERNOTIF ||
+                    self == all::OP_INVALIDOPCODE {
+                    Class::IllegalOp
+                // 11 opcodes
+                } else if self == all::OP_NOP ||
+                        (all::OP_NOP1.code <= self.code &&
+                        self.code <= all::OP_NOP10.code) {
+                    Class::NoOp
+                // 87 opcodes
+                } else if self.code == 80 || self.code == 98 ||
+                        (self.code >= 126 && self.code <= 129) ||
+                        (self.code >= 131 && self.code <= 134) ||
+                        (self.code >= 137 && self.code <= 138) ||
+                        (self.code >= 141 && self.code <= 142) ||
+                        (self.code >= 149 && self.code <= 153) ||
+                        (self.code >= 187 && self.code <= 254) {
```

Comment:
Nit: `126..=129.contains(self.code) ...` is more idiomatic (see my commend below)